src/OpenLoco/src/Ui/ToolManager.h

```diff
@@ -1,20 +1,98 @@
 #pragma once
 
 #include "Types.hpp"
+#include "Widget.h"
 #include "Window.h"
 #include <cstdint>
 
 namespace OpenLoco::ToolManager
 {
+    using namespace Ui;
+
+    enum class ToolEventType : uint8_t
+    {
+        onMouseMove,
+        onMouseDown,
+        onMouseDrag,
+        onMouseDragEnd,
+        onAbort,
+        onShiftChanged,
+        onControlChanged,
+        onScrollNoModifier,
+        onScrollShiftModifier,
+        onScrollControlModifier,
+        getCursor,
+        count
+    };
+
+    struct ToolState
+    {
+        ToolEventType event;
+        Point pos;
+        int16_t scrollWheelChanged;
+        bool shiftPressed;
```

Comment:
not really necessary to have shiftPressed and controlPressed since this is called after the bitfield is modified, so they can use the regular input method